pmlc/conversion/gpu/gpu_to_vulkan.cc

```diff
@@ -109,15 +126,61 @@ class ConvertGpuLaunchFuncToVulkanCalls
   }
 
   const char *getBufferBindingFunc(Type &elementType) {
+    if (elementType.isInteger(8)) {
+      return kBindBufferInteger8;
+    }
+    if (elementType.isInteger(16)) {
+      return kBindBufferInteger16;
+    }
+    if (elementType.isInteger(32)) {
+      return kBindBufferInteger32;
+    }
     if (elementType.isInteger(64)) {
-      return kBindBufferInt64;
+      return kBindBufferInteger64;
+    }
+    if (elementType.isBF16()) {
+      return kBindBufferBFloat16;
+    }
+    if (elementType.isF16()) {
+      return kBindBufferFloat16;
     }
     if (elementType.isF32()) {
       return kBindBufferFloat32;
     }
+    if (elementType.isF64()) {
+      return kBindBufferFloat64;
+    }
     return nullptr;
   }
 
+  void getElementTypeSize(Type &elementType, uint32_t &size) {
```

Comment:
I'm not sure why we need this method. Isn't there a `type.getIntOrFloatBitWidth()` that you can use?

Reply:
Thanks for pointing to this function, I will replace it by this function.